extensions/sql-database-projects/src/test/project.test.ts

```diff
@@ -225,7 +225,9 @@ describe('Project: sqlproj content operations', function (): void {
 		projFilePath = await testUtils.createTestSqlProjFile(baselines.newProjectFileBaseline);
 		const project = await Project.openProject(projFilePath);
 
-		await project.changeTargetPlatform('invalidPlatform');
```

Comment:
@kisantia seems like you sneaked in the change before me... The question I have is whether any of the validation builds run these tests? I'd expect this test to fail after your change as `changeTargetPlatform` used to throw an exception in this case and as soon as you add `await` here - it will be an unexpected exception. Did you also change the logic in `changeTargetPlatform` not to throw (meaning test will now fail with my changed code) or how did the merge go through the validation?

Reply:
Okay, so apparently `CI / linux` build runs extension tests and it is indeed now failing with my change because you changed the behavior and it doesn't throw an error any longer when set to incorrect platform. Was this intentional?
Looking at your changeset, this new behavior seems to come from the `getSystemDacpacUri`/`getSystemDacpacSsdtUri`, where you changed
```typescript
let version = this.getProjectTargetVersion();
```
to
```typescript
const versionFolder = this.getSystemDacpacFolderName();
```
and `getProjectTargetVersion` is what used to throw.

Can you please confirm what is the expected behavior for `changeTargetPlatform` when incorrect platform is provided?

Reply:
I moved the Telemetry event in `changeTargetPlatform()` since it was sending the wrong from target version. Before the telemetry event was added, `changeTargetPlatform()` was not throwing when an invalid target platform was passed in and it was `getProjectTargetVersion()` that throws an error if the project version is invalid. This was done this way so that tests could easily set an invalid target platform and test the scenario of when a sqlproj has an invalid target platform.

Reply:
Okay so what I'm hearing is that desired behavior is not to throw from `changeTargetPlatform` on invalid platforms, so I will update tests in my PR accordingly without going into another round of CS101, explaining why behavior of production code should not be driven by tests. 🤦